Source/OxyPlot/Series/HeatMapSeries.cs

```diff
@@ -172,6 +187,58 @@ public void Invalidate()
         }
 
         /// <summary>
+        /// Gets the clipping rectangle, transposed if the X acis is vertically orientated
+        /// </summary>
+        protected new OxyRect GetClippingRect()
+        {
+            double minX = Math.Min(this.XAxis.ScreenMin.X, this.XAxis.ScreenMax.X);
+            double minY = Math.Min(this.YAxis.ScreenMin.Y, this.YAxis.ScreenMax.Y);
+            double maxX = Math.Max(this.XAxis.ScreenMin.X, this.XAxis.ScreenMax.X);
+            double maxY = Math.Max(this.YAxis.ScreenMin.Y, this.YAxis.ScreenMax.Y);
+
+            if (XAxis.IsVertical())
+                return new OxyRect(minY, minX, maxY - minY, maxX - minX);
+            else
+                return new OxyRect(minX, minY, maxX - minX, maxY - minY);
+        }
+
+        /// <summary>
+        /// Transposes the ScreenPoint if the X axis is vertically orientated
+        /// </summary>
+        private ScreenPoint Orientate(ScreenPoint point)
+        {
+            if (XAxis.IsVertical())
+            {
+                point = new ScreenPoint(point.Y, point.X);
+            }
+            return point;
+        }
+
+        /// <summary>
+        /// Transforms DataSpace coords to orientated ScreenSpace coors
+        /// </summary>
+        public new ScreenPoint Transform(double x, double y)
+        {
+            return Orientate(base.Transform(x, y));
+        }
+
+        /// <summary>
+        /// Transforms DataSpace coords to orientated ScreenSpace coors
+        /// </summary>
+        public new ScreenPoint Transform(DataPoint p)
+        {
+            return Orientate(base.Transform(p));
+        }
+
+        /// <summary>
+        /// Transforms orientated ScreenSpace coords to DataSpace coors
+        /// </summary>
+        public new DataPoint InverseTransform(ScreenPoint point)
+        {
+            return base.InverseTransform(Orientate(point));
+        }
```

Comment:
Another node, is that while the HeatMap labels end up in the right place, trying to select them (in a PDF) results in there underlying ordering becoming apparent - this would suggest that the approach of simply repositioning them is not adequate, and for things like text more careful measures may be warranted.